Podman remote crashes when podmate-remote.conf file is incorrect

/kind bug

**Description**

Podman-remote checks podman-remote.conf for credentials saved in this conf file. However, when the conf file is not set up correctly, i.e. missing a field, the program crashes really bad. Probably a parsing and catching issue.

**Steps to reproduce the issue:**

1.  Create podman-remote.conf in ~/.conf/containers

2. In podman-remote.conf, write 
[connections]
                  [connections.host1]
                  username = "user"
                  default = true
purposely leaving out ip destination

**Describe the results you received:**
Bunch of debug panic junk
```
panic: runtime error: invalid memory address or nil pointer dereference

[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1c3c678]


goroutine 1 [running]:

github.com/containers/libpod/pkg/adapter.RemoteRuntime.RemoteEndpoint(0x0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)

	/home/ashleycui/go/src/github.com/containers/libpod/pkg/adapter/client.go:39 +0x198

github.com/containers/libpod/pkg/adapter.RemoteRuntime.Connect(0x0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)

	/home/ashleycui/go/src/github.com/containers/libpod/pkg/adapter/client.go:61 +0x55

github.com/containers/libpod/pkg/adapter.GetRuntime(0x21c3140, 0xc0000b2028, 0x2ea1780, 0xc000800480, 0x1deae20, 0x1)

	/home/ashleycui/go/src/github.com/containers/libpod/pkg/adapter/runtime_remote.go:83 +0x373

main.psCmd(0x2ea1780, 0x0, 0x0)

	/home/ashleycui/go/src/github.com/containers/libpod/cmd/podman/ps.go:214 +0xb6

main.glob..func10(0x2e78d20, 0x2ebedc0, 0x0, 0x0, 0x0, 0x0)

	/home/ashleycui/go/src/github.com/containers/libpod/cmd/podman/ps.go:164 +0x86

github.com/containers/libpod/vendor/github.com/spf13/cobra.(*Command).execute(0x2e78d20, 0xc0000ac010, 0x0, 0x0, 0x2e78d20, 0xc0000ac010)

	/home/ashleycui/go/src/github.com/containers/libpod/vendor/github.com/spf13/cobra/command.go:762 +0x465

github.com/containers/libpod/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x2e81940, 0xc00031c000, 0x7ffeefbffc88, 0x14)

	/home/ashleycui/go/src/github.com/containers/libpod/vendor/github.com/spf13/cobra/command.go:852 +0x2ec

github.com/containers/libpod/vendor/github.com/spf13/cobra.(*Command).Execute(...)

	/home/ashleycui/go/src/github.com/containers/libpod/vendor/github.com/spf13/cobra/command.go:800

main.main()

	/home/ashleycui/go/src/github.com/containers/libpod/cmd/podman/main.go:143 +0x88
```
**Describe the results you expected:**
A prompt for the user to enter ip destination

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:            1.3.2-dev

RemoteAPI Version:  1

Go Version:         go1.12.2

Git Commit:         c5c73794b4bdb13746d0cceeb2ed62583b2c46a2

Built:              Thu Jun  6 14:47:34 2019

OS/Arch:            darwin/amd64
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

macOS Mojave 10.14.5

--- a/cmd/podman/remoteclientconfig/configfile.go
+++ b/cmd/podman/remoteclientconfig/configfile.go
@@ -22,7 +22,7 @@ func ReadRemoteConfig(reader io.Reader) (*RemoteConfig, error) {
 	// We need to validate each remote connection has fields filled out
 	for name, conn := range remoteConfig.Connections {
 		if len(conn.Destination) < 1 {
-			return nil, errors.Errorf("connection %s has no destination defined", name)
+			return nil, errors.Errorf("connection %q has no destination defined", name)
 		}
 	}
 	return &remoteConfig, err

--- a/pkg/adapter/client.go
+++ b/pkg/adapter/client.go
@@ -15,8 +15,10 @@ import (
 var remoteEndpoint *Endpoint
 
 func (r RemoteRuntime) RemoteEndpoint() (remoteEndpoint *Endpoint, err error) {
-	remoteConfigConnections, _ := remoteclientconfig.ReadRemoteConfig(r.config)
-
+	remoteConfigConnections, err := remoteclientconfig.ReadRemoteConfig(r.config)
+	if errors.Cause(err) != remoteclientconfig.ErrNoConfigationFile {
+		return nil, err
+	}
 	// If the user defines an env variable for podman_varlink_bridge
 	// we use that as passed.
 	if bridge := os.Getenv("PODMAN_VARLINK_BRIDGE"); bridge != "" {
@@ -46,6 +48,10 @@ func (r RemoteRuntime) RemoteEndpoint() (remoteEndpoint *Endpoint, err error) {
 		}
 		if err != nil {
 			return nil, err
+		}
+		if len(rc.Username) < 1 {
+			logrus.Debugf("Connection has no username, using current user %q", r.cmd.RemoteUserName)
+			rc.Username = r.cmd.RemoteUserName
 		}
 		remoteEndpoint, err = newBridgeConnection("", rc, r.cmd.LogLevel)
 		//	last resort is to make a socket connection with the default varlink address for root user

